### PR TITLE
Modify git_diff_stats_to_buf to not show "0 insertions" or "0 deletions"

### LINKED ISSUE #3788: git_diff_stats_to_buf should not show "0 insertions" or "0 deletions"
`git diff --stat` (or `git diff --shortstat`) simplifies the diffstat by not showing "0 insertions" or "0 deletions".  For instance, for a patch adding a line to a file, git shows:

```
 a | 1 +
 b | 1 +
 c | 1 +
 3 files changed, 3 insertions(+)
```

For the same patch, libgit2's `git_diff_stats_to_buf` shows:

```
 a | 1 +
 b | 1 +
 c | 1 +
 3 files changed, 3 insertions(+), 0 deletions(-)
```

Please consider adding the same simplification to `git_diff_stats_to_buf`.

### PR DESCRIPTION
**Pull request by meatcoder**
This is an attempt to fix #3788. 

It attempts to mimic the output from `git diff --stat` and `git diff --shortstat` by not including the indications for "0 deletions" and "0 insertions". Note that it will show "0 insertions(+), 0 deletions(-)" for the case, where both the number of lines inserted and deleted are 0, to match how `git diff --stat` behaves in that case. 

I've run the existing tests locally and they seem to be all passing. I've also added some new tests specifically for the changes made in this PR.

### CODE REVIEW

**Comment by meatcoder** [comment]
Pushed an additional commit to fix the failing tests in diff::format::email

**Comment by meatcoder** [comment]
Push some formatting and style fixes to better match the convention. But I think this should be ready for review.

**Comment by joshtriplett** [comment]
This looks great. Thanks!

File: src/diff_stats.c [hunk 1]
```diff
@@ -300,13 +300,31 @@ int git_diff_stats_to_buf(
...
-			out, " %" PRIuZ " file%s changed, %" PRIuZ
-			" insertion%s(+), %" PRIuZ " deletion%s(-)\n",
-			stats->files_changed, stats->files_changed != 1 ? "s" : "",
-			stats->insertions, stats->insertions != 1 ? "s" : "",
-			stats->deletions, stats->deletions != 1 ? "s" : "");
+				out, " %" PRIuZ " file%s changed",
+				stats->files_changed, stats->files_changed != 1 ? "s" : "");
+		
+		if (error < 0) 
+			return error;
```
Comment (carlosmn) [inline src/diff_stats.c]: You don't need to check for errors every time. You can keep writing into the buffer and do a `git_buf_oom(out)` check in the end.

File: tests/diff/stats.c [hunk 2]
```diff
@@ -122,7 +158,7 @@ void test_diff_stats__rename(void)
 	" 2 files changed, 4 insertions(+), 1 deletion(-)\n";
 
 	diff_stats_from_commit_oid(
-		&_stats, "8947a46e2097638ca6040ad4877246f4186ec3bd", true);
+	&_stats, "8947a46e2097638ca6040ad4877246f4186ec3bd", true);
```
Comment (carlosmn) [inline tests/diff/stats.c]: This didn't change anything, did it?

**Comment by carlosmn** [comment]
Looks fine other than the small things I've pointed out.

**Comment by meatcoder** [comment]
@carlosmn thanks for the review. I'm sorry, I've neglected this PR. I am just getting back to it and I have made the changes you suggested. 

There are 2 CI failure for appveyor, but it looks like those are build time outs. If those build jobs are re-run, I believe they should pass. Let me know if there are other changes I need to make.

**Comment by ethomson** [comment]
Thanks @meatcoder !  I've squashed this manually and pushed it.  Thanks for taking care of this!

**Comment by meatcoder** [comment]
@ethomson Thanks for pushing the change 😃 .

**Pull request closed without merging**